tests/aggregation/tests.py

```diff
@@ -2116,6 +2116,19 @@ def test_multiple_aggregate_references(self):
             },
         )
 
+    def test_group_by_reference_subquery(self):
+        qs = Publisher.objects.filter(
+            pk__in=Author.objects.annotate(
+                publisher_id=F("book__publisher"),
+            )
+            .values("publisher_id")
+            .annotate(
+                cnt=Count("*"),
+            )
+            .values("publisher_id")
+        ).order_by("pk")
```

Comment:
`order_by()` is not important for this issue, so I'd remove it and use `assertCountEqual()`.